api/src/search/backend/load_opportunities_to_index.py

```diff
@@ -68,21 +72,61 @@ def run_task(self) -> None:
-        # TODO - The point of this incremental load is to support upcoming work
-        #        to load only opportunities that have changes as we'll eventually be indexing
-        #        files which will take longer. However - the structure of the data isn't yet
-        #        known so I want to hold on actually setting up any change-detection logic
-        loaded_opportunity_ids = set()
-        for opp_batch in self.fetch_opportunities():
-            loaded_opportunity_ids.update(self.load_records(opp_batch))
+        # Fetch opportunities that need processing from the queue
+        queued_opportunities = (
+            self.db_session.execute(
+                select(Opportunity)
+                .join(OpportunitySearchIndexQueue)
+                .join(CurrentOpportunitySummary)
+                .where(
+                    Opportunity.is_draft.is_(False),
+                    CurrentOpportunitySummary.opportunity_status.isnot(None),
+                )
+                .options(selectinload("*"), noload(Opportunity.all_opportunity_summaries))
+            )
+            .scalars()
+            .all()
+        )
 
-        # Delete
-        opportunity_ids_to_delete = existing_opportunity_ids - loaded_opportunity_ids
+        # Process updates and inserts
+        processed_opportunity_ids = set()
+        opportunities_to_index = []
 
-        if len(opportunity_ids_to_delete) > 0:
+        for opportunity in queued_opportunities:
+            logger.info(
+                "Processing queued opportunity",
+                extra={
+                    "opportunity_id": opportunity.opportunity_id,
+                    "in_search": opportunity.opportunity_id in existing_opportunity_ids,
+                },
+            )
+
+            # Add to index batch if it's indexable
+            opportunities_to_index.append(opportunity)
+            processed_opportunity_ids.add(opportunity.opportunity_id)
+
+        # Bulk index the opportunities (handles both inserts and updates)
+        if opportunities_to_index:
+            loaded_ids = self.load_records(opportunities_to_index)
+            logger.info(f"Indexed {len(loaded_ids)} opportunities")
+
+        # Handle deletes - opportunities in search but not in our processed set
+        # and not in our database (or are drafts)
+        opportunity_ids_to_delete = existing_opportunity_ids - processed_opportunity_ids
+
+        if opportunity_ids_to_delete:
+            logger.info(f"Deleting {len(opportunity_ids_to_delete)} opportunities from search")
             self.search_client.bulk_delete(self.index_name, opportunity_ids_to_delete)
 
+        # Clear processed entries from the queue
+        if processed_opportunity_ids:
+            self.db_session.execute(
+                delete(OpportunitySearchIndexQueue).where(
+                    OpportunitySearchIndexQueue.opportunity_id.in_(processed_opportunity_ids)
+                )
+            )
```

Comment:
WIth the queue table, we had the `has_update` column we could flip rather than deleting the records.

Virtually the same behavior, updating just gives us a some auditability (which the logs could also give, so not a requirement). Happy to go with this approach, but we might as well remove that column to avoid confusion in that case.

Reply:
I was thinking delete to avoid an endlessly growing table which might cause unknown side effects down the road. Will remove the column!

Reply:
Just flagging when troubleshooting search data not updating it can be very helpful to see a history of when we thought the data changesd and when those changes purportedly updated to search. So having like data_updated and search_updated timestamps could be really helpful as it's way easier to consume that data directly than parse it out of logs both for like ongoing metrics (how many search updates and lag of data to search change) and troubleshooting. It would create the need to clean up the table but with the platform framework for scheduling jobs having a daily or weekly job that turns anything over 30, 60, 90 days or 6 months or whatever should be pretty doable? We don't have to change this round necessarily but should think through it this is valuable to do in a follow up.

Reply:
One thing that would help and go along with what we have here. Add something like:
```
for opportunity_id in opportunity_ids_to_delete:
     logger.info("Deleting opportunity from search", extra={"opportunity_id": opportunity_id, "status": "delete")
```
It adds a lot of logs (but we already do that for inserts/updates, so whatever), but also makes finding what happened all exist in one place.

Reply:
I added this and makes sense since we can log and report on those logs later to track, and without storing these in an ever-growing table which would need an extra task to prune.